python/metatensor-learn/metatensor/learn/selection/_selection.py

```diff
@@ -0,0 +1,248 @@
+from typing import Type, Union
+
+import numpy as np
+import skmatter._selection
+
+import metatensor
+
+from .._backend import Labels, TensorBlock, TensorMap
+
+
+class GreedySelector:
+    """
+    Wraps :py:class:`skmatter._selection.GreedySelector` for a TensorMap.
+
+    The class creates a selector for each block. The selection will be done based the
+    values of each :py:class:`TensorBlock`. Gradients will not be considered for the
+    selection.
+    """
+
+    def __init__(
+        self,
+        selector_class: Type[skmatter._selection.GreedySelector],
+        selection_type: str,
+        n_to_select: Union[int, dict],
+        **selector_arguments,
+    ) -> None:
+        self._selector_class = selector_class
+        self._selection_type = selection_type
+        self._n_to_select = n_to_select
+        self._selector_arguments = selector_arguments
+
+        self._selector_arguments["selection_type"] = self._selection_type
+        self._support = None
+        self._select_distance = None
+
+    @property
+    def selector_class(self) -> Type[skmatter._selection.GreedySelector]:
+        """
+        The class to perform the selection. Usually one of 'FPS' or 'CUR'.
+        """
+        return self._selector_class
+
+    @property
+    def selection_type(self) -> str:
+        """
+        Whether to choose a subset of columns ('feature') or rows ('sample').
+        """
+        return self._selection_type
+
+    @property
+    def selector_arguments(self) -> dict:
+        """
+        Arguments passed to the ``selector_class``.
+        """
+        return self._selector_arguments
+
+    @property
+    def support(self) -> TensorMap:
+        """
+        TensorMap containing the support.
+        """
+        if self._support is None:
+            raise ValueError("No selections. Call fit method first.")
+
+        return self._support
+
+    @property
+    def get_select_distance(self) -> TensorMap:
+        """
+        Returns a TensorMap containing the Hausdorff distances.
```

Comment:
It is. I assume it is the same in skmatter but we can move this only the FPS classes or create a new base class that only the fps selectors inherit to avoid duplicating this into two classes.